Dockerfile

```diff
@@ -512,5 +512,12 @@ ADD patches/kaggle_gcp.py /root/.local/lib/python3.6/site-packages/kaggle_gcp.py
 ADD patches/kaggle_secrets.py /root/.local/lib/python3.6/site-packages/kaggle_secrets.py
 ADD patches/sitecustomize.py /root/.local/lib/python3.6/site-packages/sitecustomize.py
 
+# TensorBoard Jupyter extension
+ENV JUPYTER_CONFIG_DIR "/root/.jupyter/"
+RUN pip install jupyter_tensorboard && \
+    jupyter serverextension enable jupyter_tensorboard && \
+    jupyter tensorboard enable
+ADD patches/tensorboard/notebook.py /opt/conda/lib/python3.6/site-packages/tensorboard/notebook.py
```

Comment:
Does this overwrite an existing file, or could it in the future?  If so, how is this correct?  (Could we miss out on functionality otherwise present there?)

Reply:
It doesn't currently, but will once TensorBoard releases their extension. The idea is to provide our implementation using the server extension using the same magic commands in the TensorFlow tutorials. It's possible that they'll add more parameters at some point, but should be safe (will behave like an older version of the magic) and simple enough to update.

Reply:
Could we verify if the file exists (in an earlier `RUN` command) and fail the BUILD if so?  Would be less fragile / more apparent in that case.

Can you also mention this in file comments -- it is good to know this is only a temporary need (IIUC).

Reply:
Sorry, I wasn't clear. TensorBoard's upcoming implementation doesn't include proxying, it simply tries to connect to `localhost`, so we actually want to continue overriding their magic's behavior even when they do release it.

Reply:
k.  Would it make sense to provide the support we need upstream?  Ultimately, I see these things as a temporary solution until that is the case.  At least for now create an issue for it and refer to it in a comment here?

Reply:
I don't think it's feasible to have the magic support proxying, as the implementation would need to be specific to a frontend (e.g., JupyterLab, Jupyter, etc.). We should be able to replace this once we have worker tunneling set up: https://github.com/sagemathinc/cocalc/wiki/TensorBoard

I added a comment describing this.